src/Poisson2DSparseMatrix.cpp

```diff
@@ -139,17 +145,75 @@ namespace blitzdg{
                                                                  // for '+' traces.
 
                 real_matrix_type gDxM(NGauss, Np), gDyM(NGauss, Np),
-                    gDxP(NGauss, Np), gDyP(NGauss, Np);
+                    gDxP(NGauss, Np), gDyP(NGauss, Np), gDr(NGauss, Np), gDs(NGauss, Np);
 
-                // Evaluate spatial derivatives of  Lagrange basis function at Gauss nodes
-                dg.computeDifferentiationMatrices(xLocal, yLocal, gVM, gDxM, gDyM);
-                dg.computeDifferentiationMatrices(xLocal2, yLocal2, gVP, gDxP, gDyP);
+                // Compute x and y differentiation matrices that evaluate
+                // derivatives at Gauss nodes.
+                const real_matrix_type& gInterp = gCtx.Interp();
+
+                gDr = blitz::sum(gInterp(ii, kk)*Dr(kk, jj), kk);
+                gDs = blitz::sum(gInterp(ii, kk)*Ds(kk, jj), kk);
+
+                real_vector_type xr1(NGauss), xs1(NGauss), yr1(NGauss), ys1(NGauss),
+                    J1(NGauss), rx1(NGauss), sx1(NGauss), ry1(NGauss), sy1(NGauss);
+
+                // Compute geometric factors on this element.
+                xr1 = blitz::sum(gDr(ii, jj)*xLocal(jj), jj);
+                xs1 = blitz::sum(gDs(ii, jj)*xLocal(jj), jj);
+
+                yr1 = blitz::sum(gDr(ii, jj)*yLocal(jj), jj);
+                ys1 = blitz::sum(gDs(ii, jj)*yLocal(jj), jj);
+
+                J1 = -xs1*yr1 + xr1*ys1;
+
+                rx1 = ys1/J1;
+                sx1 =-yr1/J1;
+                ry1 =-xs1/J1;
+                sy1 = xr1/J1;
+
+                real_matrix_type gDx1(NGauss, Np), gDy1(NGauss, Np);
+                gDx1 = 0.; gDy1 = 0.;
+
+                for (index_type i=0; i < NGauss; ++i) {
+                    for (index_type j=0; j < Np; ++j) {
+                        gDx1(i, j) = rx1(i) * gDr(i, j) + sx1(i) * gDs(i, j);
+                        gDy1(i, j) = ry1(i) * gDr(i, j) + sy1(i) * gDs(i, j);
+                    }
+                }
+
+                real_vector_type xr2(NGauss), xs2(NGauss), yr2(NGauss), ys2(NGauss),
+                    J2(NGauss), rx2(NGauss), sx2(NGauss), ry2(NGauss), sy2(NGauss);
+
+                // Compute geometric factors on the neighboring element.
+                xr2 = blitz::sum(gDr(ii, jj)*xLocal2(jj), jj);
+                xs2 = blitz::sum(gDs(ii, jj)*xLocal2(jj), jj);
+
+                yr2 = blitz::sum(gDr(ii, jj)*yLocal2(jj), jj);
+                ys2 = blitz::sum(gDs(ii, jj)*yLocal2(jj), jj);
+
+                J2 = -xs1*yr1 + xr1*ys1;
+
```

Comment:
Same question as above for `J1`, can `J2` be zero and do we need to handle that case?

Reply:
Ditto explanation as J1.